### ISSUE TITLE
Emails Show New Version Available - Web UI Says Up-to-Date

### ISSUE THREAD

**Issue by jimmybrancaccio** [jokob-sk/Pi.Alert#458]
## Describe the issue

> When submitting an issue ❗[enable debug](https://github.com/jokob-sk/Pi.Alert/blob/main/docs/DEBUG_TIPS.md)❗ and [have a look at the docs](https://github.com/jokob-sk/Pi.Alert/tree/main/docs)

When I get emails from PiAlert they show that a new version is available. Upon logging into the web UI and going to the Maintenance page I see that it is up-to-date. I search through GitHub issues and didn't see anyone else report this. I would normally spin up a dev instance, but as I only run one instance I didn't want any database issues if/when I revert back to the production build.

## Paste your `pialert.conf` (remove personal info)

I don't believe this is necessary but please let me know if you'd like it.

## Paste your `docker-compose.yml` and `.env` (remove personal info)

`docker-compose.yml` 

```yaml
pialert:
    image: jokobsk/pi.alert:latest
    container_name: pialert
    labels:
      - traefik.enable=true
      - traefik.docker.network=traefik-public
      - traefik.constraint-label=traefik-public
      - traefik.http.routers.pialert.rule=Host(`pialert.mydomain.com`)
      - traefik.http.routers.pialert.entrypoints=websecure
      - traefik.http.routers.pialert.middlewares=authelia@docker
      - traefik.http.routers.pialert.service=pialert
      - traefik.http.routers.pialert.tls=true
      - traefik.http.routers.pialert.tls.certresolver=letsencrypt
      - traefik.http.routers.pialert.tls.domains[0].main=pialert.mydomain.com
      - traefik.http.services.pialert.loadbalancer.server.port=20211
    environment:
      - TZ=${TZ}
      - HOST_USER_ID=${PUID}
      - HOST_USER_GID=${PGID}
    volumes:
      - ${DOCKER_DATA_DIR}/pialert/db:/home/pi/pialert/db
      - ${DOCKER_DATA_DIR}/pialert/config:/home/pi/pialert/config
      - ${DOCKER_LOGS_DIR}/pialert:/home/pi/pialert/front/log
      - ${DOCKER_DATA_DIR}/pialert/api:/home/pi/pialert/front/api
      - ${DOCKER_DATA_DIR}/pialert/font-awesome:/home/pi/pialert/front/lib/AdminLTE/bower_components/font-awesome:ro
    network_mode: host
    restart: unless-stopped
```

`.env` 

```
TZ=America/Chicago
PUID=1000
PGID=1000
DOCKER_DATA_DIR=/mnt/data/docker/data
DOCKER_LOGS_DIR=/mnt/data/docker/logs
```

## Screenshots

[If applicable, add screenshots to help explain your problem.]

<img width="776" alt="Screenshot 2023-09-20 at 11 43 39" src="https://github.com/jokob-sk/Pi.Alert/assets/405585/6d7425d2-caeb-42b2-a796-c77b2f348d09">

<img width="868" alt="Screenshot 2023-09-29 at 11 14 03" src="https://github.com/jokob-sk/Pi.Alert/assets/405585/be34288a-f4fb-4f3b-bb5d-28f1259dd046">

## Paste last few lines from `pialert.log`

> You can use `tail -100 /home/pi/pialert/front/log/pialert.log`

```
11:25:57 [Internet IP] Check Internet IP started
11:25:57 [Internet IP] - Retrieving Internet IP
11:25:57 [Internet IP] IP:      1.2.3.4
11:25:57 [Internet IP]    Retrieving previous IP:
11:25:57 [Internet IP]      1.2.3.4
11:25:57 [Internet IP]    No changes to perform
11:25:57 [DDNS]     Skipping Dynamic DNS update
11:25:57 [Notification] Check if something to report
11:25:57 [Notification] Open text Template
11:25:57 [Notification] Open html Template
11:25:57 [Notification] Using template/back/report_template_new_version.html
11:25:57 [Notification] included sections: ['internet', 'new_devices', 'down_devices', 'events']
11:25:57 [Notification] Internet sections done.
11:25:57 [Notification] New Devices sections done.
11:25:57 [Notification] Down Devices sections done.
11:25:57 [Notification] Events sections done.
11:25:57 [Notification] No changes to report
11:25:57 [Notification] Notifications changes: 0
11:25:57 [MAIN] Process: Wait
11:26:02 [MAIN] waiting to start next loop
11:26:07 [MAIN] waiting to start next loop
11:26:12 [MAIN] waiting to start next loop
11:26:17 [MAIN] waiting to start next loop
11:26:22 [MAIN] waiting to start next loop
11:26:27 [MAIN] waiting to start next loop
11:26:33 [MAIN] waiting to start next loop
11:26:38 [MAIN] waiting to start next loop
```

**Comment by jokob-sk** [comment]
HI there,

Thanks for the detailed description. This helps a lot! 🙏

It seems like you are not on the latest version, the latest version is built on 9.13.2023. See below screenshot. 

![image](https://github.com/jokob-sk/Pi.Alert/assets/96159884/814d7636-523a-43e1-8daa-a8402c977020)

There might be a bug in the UI version detection, I will have a look into that. 

Thanks,
j

**Comment by jimmybrancaccio** [comment]
Hrmm, so this is weird then:

```console
docker pull jokobsk/pi.alert:latest

latest: Pulling from jokobsk/pi.alert
Digest: sha256:dc893f509b1eb668bcf21fc61ca5a5b1d1f1468fced0ca5a8b76e7749b1c3290
Status: Image is up to date for jokobsk/pi.alert:latest
docker.io/jokobsk/pi.alert:latest
```

I even deleted the container and re-created it. 🤔 Looking at https://hub.docker.com/r/jokobsk/pi.alert/tags it shows latest was last pushed 17 days ago which would be the 12.

You've got some magic going on in your Maintenance page 😝

**Comment by jokob-sk** [comment]
I think the issue is in the build date vs. build tag. It's difficult to get the build tag into the image when building it so I rely on the build date. There were 3 releases in short succession (one every day) so in this case, this might lead to some confusion. 

I'm unifying the version check for the UI and backend so those should be aligned in the next release, which should prevent conflicting notifications. Still, in your case, I think you are not running the latest image.

**Comment by jimmybrancaccio** [comment]
> Still, in your case, I think you are not running the latest image.

I think you're definitely right. I just deleted the container again, removed the image and created a new container where `docker compose` pulled the image. It still seems to be using the image from the 12th haha Inspecting the image the container is using I also see:

```
org.opencontainers.image.created  - 2023-09-12T22:12:59.908Z
org.opencontainers.image.version - 23.9.12
```

<img width="1249" alt="Screenshot 2023-09-29 at 17 22 43" src="https://github.com/jokob-sk/Pi.Alert/assets/405585/dcc84b9b-3781-4d51-bb3d-cdb7a278a919">

So I don't know what is going on lol

**Comment by jokob-sk** [comment]
Thanks for checking. I will probably have this fixed in the `_dev` image in the next couple of hours if you'd like to test it later. The app will rely on a global app state that is updated in the backend so there will be no discrepancies - if there is a bug, it will be only in one place. :)

**Comment by tismofied** [comment]
I can confirm also I am on 23.9.12 still. I even added latest tag at the end of the pull link. Hopefully fixed in the next release.

**Comment by jokob-sk** [comment]
closing as released in v23.10.2
